### ISSUE TITLE
2048x1024 texture cannot be imported (3D).

### ISSUE THREAD

**Issue by leiget** [godotengine/godot#15139]
**Godot version:**
v3.0.beta.custom_build.f11a138

**OS/device including version:**
Xubuntu 16.04

**Issue description:**
2048x1024 size texture either stalls indefinitely during import, or simply crashes after a couple of seconds (when trying to import into 3D, ultimately). It can import a 2049x1024 image and a 2048x1025 one, as well.

**Steps to reproduce:**
Simply try to import a 2048x1024 texture onto a 3D model.

**Minimal reproduction project:**
Here are a couple of images I'm having a problem with.
This one, below this line, is simply a white test texture.
![test](https://user-images.githubusercontent.com/20325471/34427653-479909ea-ec0a-11e7-826f-3c53766b9532.png)
![cedarlebaneseplanks_01_c](https://user-images.githubusercontent.com/20325471/34427669-62983306-ec0a-11e7-8bd8-d9fda5bf2f46.png)

- [x] Yes, I searched the existing [GitHub issues](https://github.com/godotengine/godot/issues?utf8=%E2%9C%93&q=is%3Aissue+) for potential duplicates.

**Comment by reduz** [comment]
I can't reproduce.. any more information or context, or example project would be welcome

**Comment by leiget** [comment]
Okay, here ya go:
https://drive.google.com/open?id=1IKq1C5-NC8N7DzjOa8AqWw_5utVNvSbr

It was happening with any 2048x1024 texture on any project and any 3D object, like when I would try dragging the texture from the Godot browser into a texture slot, like Albedo or Normalmap. In this sample project I can import the 2050x1025 texture without issue, but not the 2048x1024 texture.

**Comment by eon-s** [comment]
With the example project I get the editor stuck on the re-import process when assigning at albedo (but I see the texture applied), after a few seconds it completes the process and continues working normally.

On the terminal I get the error:
```
ERROR: _compress_etc: Condition ' next_power_of_2(imgw) != imgw || next_power_of_2(imgh) != imgh ' is true.
   At: modules/etc/image_etc.cpp:121.
begin encoding, format: ETC2_RGB8
```

**Comment by leiget** [comment]
@eon-s I tried it on Win7 64-bit and the same thing happens as you just explained. I guess on by Xubuntu 16.04 64-bit, for some reason it doesn't work. I'll try finding an error code on Linux here in a minute.

**Comment by eon-s** [comment]
I have tested on Ubuntu 17.10 (gnome) and proprietary nvidia drivers (if that matters).

**Comment by leiget** [comment]
Here's the error.

```
ERROR: _compress_etc: Condition ' next_power_of_2(imgw) != imgw || next_power_of_2(imgh) != imgh ' is true.
   At: modules/etc/image_etc.cpp:121.
begin encoding, format: ETC2_RGB8
godot.x11.opt.tools.64: thirdparty/etc2comp/EtcBlock4x4Encoding_ETC1.cpp:792: void Etc::Block4x4Encoding_ETC1::TryDifferentialHalf(Etc::DifferentialTrys::Half*): Assertion `ptry->m_fError < FLT_MAX' failed.
Aborted (core dumped)
```

**Comment by leiget** [comment]
Even after the fix above (710970f), it was still crashing after importing a 2048x1024 texture. But then I found out that the problem was I was using unsafe optimizations while compiling, "-ffast-math" to be exact. After I got rid of that flag I'm able to import without issue.
